test/integration/datasets/DatasetsRepository.test.ts

```diff
@@ -48,48 +55,130 @@ describe('DatasetsRepository', () => {
   describe('getAllDatasetPreviews', () => {
     const testPageLimit = 1
     const expectedTotalDatasetCount = 4
+    let firstDatasetIds: CreatedDatasetIdentifiers
+    let secondDatasetIds: CreatedDatasetIdentifiers
+    let thirdDatasetIds: CreatedDatasetIdentifiers
+    let fourthDatasetIds: CreatedDatasetIdentifiers
+
+    beforeAll(async () => {
+      await createCollection()
+      await createDatasets()
+    })
+
+    afterAll(async () => {
+      await deleteDatasets()
+      await deleteCollection()
+    })
+
+    const createCollection = async () => {
+      try {
+        await createCollectionViaApi(TestConstants.TEST_CREATED_COLLECTION_ALIAS_1)
+      } catch (error) {
+        throw new Error('Tests beforeAll(): Error while creating test collection')
+      }
+    }
+
+    const createDatasets = async () => {
+      try {
+        firstDatasetIds = await createDataset.execute(
+          TestConstants.TEST_NEW_DATASET_DTO,
+          TestConstants.TEST_CREATED_COLLECTION_ALIAS_1
+        )
+        secondDatasetIds = await createDataset.execute(
+          TestConstants.TEST_NEW_DATASET_DTO,
+          TestConstants.TEST_CREATED_COLLECTION_ALIAS_1
+        )
+        thirdDatasetIds = await createDataset.execute(
+          TestConstants.TEST_NEW_DATASET_DTO,
+          TestConstants.TEST_CREATED_COLLECTION_ALIAS_1
+        )
+        fourthDatasetIds = await createDataset.execute(
+          TestConstants.TEST_NEW_DATASET_DTO,
+          TestConstants.TEST_CREATED_COLLECTION_ALIAS_1
+        )
+
+        await waitForDatasetsIndexedInSolr(
+          expectedTotalDatasetCount,
+          TestConstants.TEST_CREATED_COLLECTION_ALIAS_1
+        )
+      } catch (error) {
+        throw new Error('Tests beforeAll(): Error while creating test datasets')
+      }
+    }
+
+    const deleteDatasets = async () => {
+      try {
+        await deleteUnpublishedDatasetViaApi(firstDatasetIds.numericId)
+        await deleteUnpublishedDatasetViaApi(secondDatasetIds.numericId)
+        await deleteUnpublishedDatasetViaApi(thirdDatasetIds.numericId)
+        await deleteUnpublishedDatasetViaApi(fourthDatasetIds.numericId)
+      } catch (error) {
+        throw new Error('Tests afterAll(): Error while deleting test datasets')
+      }
+    }
```

Comment:
I suggest storing the dataset IDs in an array, instead of in separate variables. This way, we can create the datasets using a loop, with each array element representing a dataset ID in sequence (first, second, etc.).

Then, for the deleteDatasets function, you can simply pass the array of IDs and use a loop to delete the datasets, rather than calling the delete function individually for each one